library/src/main/java/eu/kanade/tachiyomi/source/ResolvableSource.kt

```diff
@@ -0,0 +1,42 @@
+package eu.kanade.tachiyomi.source
+
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import okhttp3.HttpUrl
+
+/**
+ * Represents a source that can resolve a URL into a specific type of [Resource].
+ */
+interface ResolvableSource {
+
+    /**
+     * Checks whether the given [url] can be resolved by this source.
+     *
+     * @param url The [HttpUrl] to test.
+     * @return `true` if this source can resolve the URL, `false` otherwise.
+     */
+    fun canResolve(url: HttpUrl): Boolean
+
+    /**
+     * Resolves the given [url] into a [Resource].
+     *
+     * @param url The [HttpUrl] to resolve.
+     * @return The resolved [Resource].
+     * @throws Exception if the resource cannot be resolved.
+     */
+    suspend fun getResource(url: HttpUrl): Resource
+
+    sealed interface Resource {
+
+        /**
+         * A manga resource, optionally containing a specific chapter.
+         *
+         * @property manga The resolved manga metadata.
+         * @property chapter An optional specific chapter of the manga.
+         */
+        data class Manga(
```

Comment:
Don't use data class. The methods added are useless, and `SManga/SChapter` are not hashable either.

Reply:
Why did we add it for `MangasPage` then...

Reply:
That's an app-side implementation. Extensions should not rely on this. That's why we need to unmark `MangasPage` in 1.6.